Program.cs

```diff
@@ -1,57 +1,396 @@
+                        Console.Write("Do you want to return to the Mode Selection Menu? (y/n): ");
+                        if (Console.ReadKey().KeyChar == 'y')
+                            valid = true;
+                        break;
+                    case "5":
+                        Console.Write("\nPlease key in a city's name and a porvinc's name seperated by a comma to display the city on a map(ex. 'London, Ontario'): ");
+                        string cityName;
+                        string provName;
+                        do
+                        {
+                            response = Console.ReadLine();
+                            string[] cityProv = response.Split(',', 2);
+                            cityName = cityProv[0];
+                            provName = cityProv[1];
+                            cityName = FormatAnswer(cityName);
+                            provName = FormatAnswer(provName);
+                            if (!cityStats.CityIsValid(cityName) || (!cityStats.ProvinceIsValid(provName)))
+                            {
+                                Console.Write("\nInvalid response. Please pick a city and province from the following list, formatted as 'City, Province': ");
+                            }
+                        } while (!cityStats.CityIsValid(cityName) || (!cityStats.ProvinceIsValid(provName)));
+                        cityStats.ShowCityOnMap(cityName, provName);
+                        Console.WriteLine("Do you want to return to the Mode Selection Menu? (y/n):");
+                        if (Console.ReadKey().KeyChar == 'y')
+                            valid = true;
+                        break;
+                    case "6":
+                        Console.Write("\nPlease key in two cities, separated by ',' to compare the population of those two cities: ");
+                        do
+                        {
+                            response = Console.ReadLine();
+                            string[] cities = response.Split(',', 2);
+                            cityOne = cities[0];
+                            cityTwo = cities[1];
+                            cityOne = FormatAnswer(cityOne);
+                            cityTwo = FormatAnswer(cityTwo);
+                            if (!cityStats.CityIsValid(cityOne) || !cityStats.CityIsValid(cityTwo))
+                            {
+                                Console.Write("\nInvalid response. Please pick from two of the following cities and key in the city name: ");
+                                cityStats.DisplayCitiesList();
+                                Console.WriteLine("Please key in two cities, separated by ',' from the above list: ");
+                            }
+                        } while (!cityStats.CityIsValid(cityOne) || (!cityStats.CityIsValid(cityTwo)));
+                        cityStats.CalculateDistanceBetweenCities(cityOne, cityTwo);// how do I call await on this void method?
```

Comment:
so async follows similar means to our JS course. so you would have to put await on it. It returns a Task so it is awaitable in the most updated version of the master branch. As a result though, this method we are in has to become an async method that returns a Task. Whatever method calls this one also has to await it, be async, and return a Task of whatever return it had before.

For example, if a method that returned a string was to implement a async/await (taskable) method, then it has to now return a Task<string> so it is awaitable.

Reply:
I believe the better way is to just call .Wait()
`
Task result = cityStats.CalculateDistanceBetweenCities(cityOne, cityTwo);
result.Wait();
`